feat: AWS security overview (#279)
canonical/ubuntu-cloud-docs: aws/aws-how-to/security/ubuntu-security-on-aws.rst

Ubuntu Security on AWS
######################

Ubuntu security features
************************

Ubuntu on AWS provides all security features available to Ubuntu Server. For more detailed information about how Ubuntu offer security, refer to the following `Introduction to Security <https://documentation.ubuntu.com/server/explanation/intro-to/security/>`_.

> AWS security features

Reviewer: we should add AMD SEV-SNP as well since its supported on 22.04+

Secure Boot and TPM
~~~~~~~~~~~~~~~~~~~

AWS Secure Boot is a feature providing secure boot process of Amazon EC2 so instance only boots trusted softwares. To create and configure secure boot image with Ubuntu AMI, check out :doc:`./use-secureboot-and-vtpm`.

CIS hardened image
~~~~~~~~~~~~~~~~~~

CIS hardened images are built with configuration by the Center for Internet Security (CIS) for use on Amazon EC2.

To create hardened image, follow instructions in `Comply with CIS or DISA STIG on Ubuntu 20.04 with Ubuntu Security Guide <https://ubuntu.com/tutorials/comply-with-cis-or-disa-stig-on-ubuntu#1-overview>`_.

Ubuntu Pro
**********

Ubuntu offers various type of AWS images to meet security requirements. Images can have enhanced security features with `Ubuntu Pro <https://ubuntu.com/aws/pro>`_.

Images with Pro can offer:

* Expanded Security Maintenance (ESM): Provide 10 years of ubuntu software package management.
* Live kernel updates: Reduce downtime and unplanned reboots for kernel vulnerabilities.
* FIPS: Configure system to meet strict security requirements creating highly regulated environments.

> To learn more about Ubuntu Pro, check out `Ubuntu Pro for AWS <https://ubuntu.com/aws/pro>`_.

Reviewer: its worth highlighting in our docs about how to find pro enabled images (specifically EKS)

Ubuntu Security on AWS
######################

Ubuntu security features
************************

> Ubuntu on AWS provides all security features available to Ubuntu Server. For more detailed information about how Ubuntu offer security, refer to the following `Introduction to Security <https://documentation.ubuntu.com/server/explanation/intro-to/security/>`_.

Reviewer 2: ```suggestion
Ubuntu on AWS provides all the security features available on Ubuntu Server. For more detailed information about how Ubuntu offers security, refer to the following `Introduction to Security <https://documentation.ubuntu.com/server/explanation/intro-to/security/>`_.
```

AWS security features
*********************

> AWS offers comprehensive security and data protection in the cloud. `Security in Amazon EC2 <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-security.html>`_ explains how user can benefit from AWS security features.

Reviewer 2: ```suggestion
AWS offers comprehensive security and data protection in the cloud. `Security in Amazon EC2 <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-security.html>`_ explains how users can benefit from AWS security features.
```

Secure Boot and TPM
~~~~~~~~~~~~~~~~~~~

> AWS Secure Boot is a feature providing secure boot process of Amazon EC2 so instance only boots trusted softwares. To create and configure secure boot image with Ubuntu AMI, check out :doc:`./use-secureboot-and-vtpm`.

Reviewer 2: ```suggestion
AWS Secure Boot is a feature of Amazon EC2 instances, that allows only trusted software to be used during the booting process. To create and configure a secure boot image using an Ubuntu AMI, refer to :doc:`./use-secureboot-and-vtpm`.
```

CIS hardened image
~~~~~~~~~~~~~~~~~~

> CIS hardened images are built with configuration by the Center for Internet Security (CIS) for use on Amazon EC2.

Reviewer 2: ```suggestion
CIS hardened images are available for use on Amazon EC2. These images include the security related configurations specified by the Center for Internet Security (CIS).
```

> To create hardened image with AWS Ubuntu Pro image, follow instructions in `Comply with CIS or DISA STIG on Ubuntu 20.04 with Ubuntu Security Guide <https://ubuntu.com/tutorials/comply-with-cis-or-disa-stig-on-ubuntu#1-overview>`_.

Reviewer 2: ```suggestion
To create a hardened image using Ubuntu Pro, follow the instructions in `Comply with CIS or DISA STIG on Ubuntu 20.04 with Ubuntu Security Guide <https://ubuntu.com/tutorials/comply-with-cis-or-disa-stig-on-ubuntu#1-overview>`_.
```
The instructions mentioned in that link only seem to [perform an audit](https://ubuntu.com/tutorials/comply-with-cis-or-disa-stig-on-ubuntu#5-run-the-ubuntu-security-guide), and don't really tell you how to harden the image. So we should either change the link to a more appropriate one, or change our text and include details for hardening. Some useful links:
- https://ubuntu.com/security/certifications/docs/usg/cis/compliance
- the video at https://ubuntu.com/security/cis

Author: I searched a little bit and Carlos presented this long time ago with Ubuntu Pro image so we can directly link to this video for tutorial.

AMD SEV-SNP
~~~~~~~~~~~

> AMD Secure Encrypted Virtualization-Secure Nested Paging (AMD SEV-SNP) provides strong memory integrity protection to instances that use AMD EPYC processors. Full details with how to launch AMD SEV-SNP instances are in :doc:`../instances/launch-and-attest-amd-sev-snp-instances`.

Reviewer 2: ```suggestion
AMD Secure Encrypted Virtualization-Secure Nested Paging (AMD SEV-SNP) provides strong memory integrity protection to instances that use AMD EPYC processors. Details about launching AMD SEV-SNP instances are given in :doc:`../instances/launch-and-attest-amd-sev-snp-instances`.
```

> Ubuntu Pro

Reviewer 2: ```suggestion
Enhanced security using Ubuntu Pro
```

**********

> Ubuntu offers various type of AWS images to meet security requirements. Images can have enhanced security features with `Ubuntu Pro <https://ubuntu.com/aws/pro>`_.

Reviewer 2: ```suggestion
Apart from the Ubuntu Server images, AWS also has images for `Ubuntu Pro <https://ubuntu.com/aws/pro>`_, which come with enhanced security features:
```

> Images with Pro can offer:

Reviewer 2: Delete this line.

> * Expanded Security Maintenance (ESM): Provide 10 years of ubuntu software package management.

Reviewer 2: ```suggestion
* Expanded Security Maintenance (ESM): Provides 10 years of security patching for packages in the Ubuntu (main and universe) repositories.
```

> * Live kernel updates: Reduce downtime and unplanned reboots for kernel vulnerabilities.

Reviewer 2: ```suggestion
* Live kernel updates: These reduce downtime and unplanned reboots in case of kernel vulnerabilities.
```

> * FIPS: Configure system to meet strict security requirements creating highly regulated environments.

Reviewer 2: ```suggestion
* FIPS compliance: Includes FIPS-certified modules to enable the use of Ubuntu in highly regulated environments.
```

> :doc:`../instances/find-ubuntu-images` explains how to find Ubuntu Pro images for AWS.

Reviewer 2: ```suggestion
To find Ubuntu Pro images on AWS (for both EC2 and EKS), refer to :doc:`../instances/find-ubuntu-images`. The product parameter allows you to specify Pro.
```

> To learn more about Ubuntu Pro, check out `Ubuntu Pro for AWS <https://ubuntu.com/aws/pro>`_.

Reviewer 2: Since this is the same link as the one specified at the beginning of this section, we can remove this line
